dluhc-component-library/src/components/siteSelectionForm/components/Input.tsx

```diff
@@ -15,14 +15,24 @@ const Input = <T extends number | string>({
   max,
   onChange,
 }: InputProps<T>) => {
+  const handleChange = (value: string) => {
+    let parsedValue: number | string = value;
+
+    if (type == "number") {
+      parsedValue = Number(value);
+    }
+
+    onChange(parsedValue as T);

```

Comment:
We should probably be using parseFloat() instead of Number(). I admit I did have to do a quick read up on the difference again though.

Some nasty stuff going on with the types here. we can probably just do
onChange(type === "number" ? parseFloat(value) : value)
I'd hope that solves the type problem but we may still need to do the as T cast because we havent done any duck typing. The above solution will at least get rid of the weird number | string that we then also cast to T though.